technology/kotlin-notebook/school-timetabling.ipynb

```diff
@@ -0,0 +1,641 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# School timetabling kotlin notebook\n",
+    "\n",
+    "This Kotlin Notebook solves the school timetabling problem with [Timefold](https://timefold.ai], the open source planning solver AI.\n",
+    "\n",
+    "![School timetabling input output](https://timefold.ai/docs/timefold-solver/latest/_images/quickstart/school-timetabling/schoolTimetablingInputOutput.png)\n",
+    "\n",
+    "## Dependencies\n",
+    "\n",
+    "Add the Timefold solver dependency:"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 74,
+   "outputs": [],
+   "source": [
+    "@file:DependsOn(\"ai.timefold.solver:timefold-solver-core:1.3.0\")"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-10-30T21:01:02.229748528Z",
+     "start_time": "2023-10-30T21:00:59.103295531Z"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "\n",
+    "## Domain\n",
+    "\n",
+    "Create the domain classes:\n",
+    "\n",
+    "### Room\n",
+    "\n",
+    "A school has rooms."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 75,
+   "outputs": [],
+   "source": [
+    "class Room {\n",
+    "\n",
+    "    var name: String\n",
+    "\n",
+    "    constructor(name: String) {\n",
+    "        this.name = name\n",
+    "    }\n",
+    "\n",
+    "    override fun toString(): String = name\n",
+    "\n",
+    "}"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-10-30T21:01:02.274546567Z",
+     "start_time": "2023-10-30T21:01:02.228839493Z"
+    }
+   }
```

Comment:
Dtto.

Reply:
I am not to worried about this kind of user data.
It's fairly similar to the git log telling when I commited the change :)
Except that it gives an indication on how old my development machine is... ;-)